src/commands/lock.rs

```diff
@@ -66,10 +76,12 @@ pub async fn lock(args: Args) -> Result<()> {
                     if let Some(runtime) = t.runtime() {
                         if let Some(container) = runtime.container() {
                             if let Ok(image) = container.value() {
-                                if let Ok(text) =
-                                    serde_json::from_str(image.expr().text().to_string().as_str())
-                                {
-                                    images.insert(text);
+                                if let Expr::Literal(LiteralExpr::String(s)) = image.expr() {
+                                    if let Some(text) = s.text() {
+                                        let mut buffer = String::new();
```

Comment:
performance tip that will never make a tangible impact: you could declare the buffer outside the `for result` loop and clear it after each `insert(buffer.clone())` so that you aren't making a bunch of one-off allocations